trio/tests/test_threads.py

```diff
@@ -164,6 +167,119 @@ async def main():
     assert record == ["sleeping", "cancelled"]
 
 
+async def test_named_thread():
+    ending = " from trio.tests.test_threads.test_named_thread"
+
+    def inner(name="inner" + ending) -> threading.Thread:
+        assert threading.current_thread().name == name
+        return threading.current_thread()
+
+    def f(name: str) -> Callable[[None], threading.Thread]:
+        return partial(inner, name)
+
+    # test defaults
+    await to_thread_run_sync(inner)
+    await to_thread_run_sync(inner, thread_name=None)
+
+    # functools.partial doesn't have __name__, so defaults to None
+    await to_thread_run_sync(f("None" + ending))
+
+    # test that you can set a custom name, and that it's reset afterwards
+    async def test_thread_name(name: str):
+        thread = await to_thread_run_sync(f(name), thread_name=name)
+        assert re.match("Trio thread [0-9]*", thread.name)
+
+    await test_thread_name("")
+    await test_thread_name("fobiedoo")
+    await test_thread_name("name_longer_than_15_characters")
+
+    await test_thread_name("💙")
+
+
+def _get_thread_name(ident: Optional[int] = None) -> Optional[str]:
+    import ctypes
+    import ctypes.util
+
+    libpthread_path = ctypes.util.find_library("pthread")
+    if not libpthread_path:
+        print(f"no pthread on {sys.platform})")
+        return None
+    libpthread = ctypes.CDLL(libpthread_path)
+
+    pthread_getname_np = getattr(libpthread, "pthread_getname_np", None)
+    # this should never fail afaik, but if so just switch the below lines
+    assert pthread_getname_np
+    # if not pthread_getname_np: # pragma: no cover
+    #    print(f"no pthread_getname_np on {sys.platform})")
+    #    return None
```

Comment:
Uncomment or delete?

Reply:
I guess line spacing didn't make it clear, but the comment on line 210 refers to 211 vs 212-214

Reply:
but probably unnecessary, updated the comment and removed the commented out lines.